### PR TITLE
ROPC added for confidential clients

### PR DESCRIPTION
**Pull request by sameerag**
Add ROPC support for `Confidential Client Applications`.

### CODE REVIEW

File: lib/msal-node/src/client/ClientApplication.ts [hunk 1]
```diff
@@ -210,6 +213,38 @@ export abstract class ClientApplication {
...
+     *
+     * @param request - UsenamePasswordRequest
+     */
+    async acquireTokenByUsernamePassword(request: UsernamePasswordRequest): Promise<AuthenticationResult | null> {
+        this.logger.info("acquireTokenByUsernamePassword called", request.correlationId);
+        const validRequest: CommonUsernamePasswordRequest = {
+            ...request,
+            ...this.initializeBaseRequest(request)
+        };
+        const serverTelemetryManager = this.initializeServerTelemetryManager(ApiId.acquireTokenByUsernamePassword, validRequest.correlationId!);
```
Comment (jasonnutter) [inline lib/msal-node/src/client/ClientApplication.ts:232]: Looks like we should turn off whatever lint rule is raising the "Forbidden non-null assertion" warning/error.

**Review by jasonnutter — APPROVED** [review]
Looks good, any docs or samples that could be updated?

File: lib/msal-node/src/client/ClientApplication.ts (hunk 1 above)
Comment (tnorling) [reply lib/msal-node/src/client/ClientApplication.ts:232]: We turned this rule on when we turned on `strictNullChecks` to remind us to fix all the non-null assertions in node. We should fix all these warnings and then set the lint rule to error

**Comment by sameerag** [comment]
> Looks good, any docs or samples that could be updated?

Yes. Working on a sample update and a few docs, will add in sometime.

File: lib/msal-node/src/client/ClientApplication.ts (hunk 1 above)
Comment (jasonnutter) [reply lib/msal-node/src/client/ClientApplication.ts:232]: Makes sense, sounds good.

Comment (sameerag) [reply lib/msal-node/src/client/ClientApplication.ts:232]: Let's log this in ADO @tnorling. All pending work under strict-null-checks.

**Review by pkanher617 — APPROVED** [review]
Should make sure we update the relevant docs inside the repo, as well as on docs.microsoft.com

**Comment by sameerag** [comment]
> Should make sure we update the relevant docs inside the repo, as well as on docs.microsoft.com

I checked and docs are already up to date. We do not add any docs in MSAL JS, but redirect it to docs.microsoft.com and we do mention that it is supported for both public and confidential clients.

**Review by samuelkubai — APPROVED** [review]
LGTM

**Pull request merged**